source/slang/check.cpp

```diff
@@ -7800,7 +7853,12 @@ namespace Slang
         {
             QualType qualType;
             qualType.type = GetType(varDeclRef);
-            qualType.IsLeftValue = true; // TODO(tfoley): allow explicit `const` or `let` variables
+
+            bool isLValue = true;
+            if(varDeclRef.getDecl()->FindModifier<ConstModifier>())
```

Comment:
On the first point, I suppose we could write it more compactly that way, but I'm also sort of anticipating a future where I add `let` and `var` keywords as alternate syntax for variable declarations, in which case we'd also have to detect a variable introduced with `let` as another non-lvalue case here. My general style is to prefer defining things in ways that make it easy to add additional cases (even if that means using mutation), rather than write large compound Boolean expressions that have to be untangled, but that is a matter of preference.

On your second point, the current situation in Slang is that `const` is a *declaration* modifier, and we don't actually have an *type* modifiers/qualifiers (unlike C/C++). This might have to change if/when we add explicit pointers, but even there it might be that we don't allow for a type like `const float` to exist, but we *do* allow for `const float *`.

Either way, if we start to add a separate notion of type modifiers then we'd need to revisit this code anyway, but certainly a check for a `const` modifier on the declaration in this case would *not* catch any `const` modifier on its type (whether top-level or nested).